src/api/deposit/DepositApi.ts

```diff
@@ -188,18 +184,25 @@ export class DepositApiImpl implements DepositApi {
 
     assert(address, `EpochTokenLocker was not deployed to network ${networkId}`)
 
-    // as string as assert is not detected by TS
-    return this._getContractAtAddress(address)
+    return this._getContractAtAddress(networkId, address)
   }
 
-  protected _getContractAtAddress(address: string): BatchExchangeContract {
-    const contract = DepositApiImpl._contractsCache[address]
```

Comment:
Interesting, we would have a collision in the previous implementation if the address where the same in rinkeby and mainnet --> it's possible if u use the same account

Reply:
We won't because I've updated the contract cache to be based on `networkId`. Check:
- https://github.com/gnosis/dex-react/blob/329/network-agnostic-apis/src/api/deposit/DepositApi.ts#L56
- https://github.com/gnosis/dex-react/blob/329/network-agnostic-apis/src/api/deposit/DepositApi.ts#L190-L207
- https://github.com/gnosis/dex-react/blob/329/network-agnostic-apis/src/api/erc20/Erc20Api.ts#L75
- https://github.com/gnosis/dex-react/blob/329/network-agnostic-apis/src/api/erc20/Erc20Api.ts#L201-L217